internal/testsupport/helpers.go

```diff
@@ -42,3 +44,39 @@ func SetupEngineMock(t *testing.T) (*mocks.MockEngine, configuration.Configurati
 	mockEngine.EXPECT().GetConfiguration().Return(engineConfig).AnyTimes()
 	return mockEngine, engineConfig
 }
+
+func ReadMessageWithFatalTimeout[T any](t *testing.T, channel chan T, timeoutDuration time.Duration) (msg T) {
+	t.Helper()
+
+	var ok bool
+	select {
+	case msg, ok = <-channel:
+		if !ok {
+			// Fatal failure (stop execution), as mishandled channels in Go can cause panics and infinite waits.
+			assert.FailNow(t, "Expected a message, but the channel was closed")
+		}
+		return msg
+	case <-time.After(timeoutDuration):
+		// Fatal failure (stop execution), as mishandled channels in Go can cause panics and infinite waits.
+		assert.FailNow(t, "Expected a message, but no message was available after %v", timeoutDuration)
+	}
+	return msg // Unreachable
+}
```

Comment:
how about using require.Eventually or assert.Eventually instead of maintaining our own code?